pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -107,3 +108,21 @@ def attach_model_logging_functions(self, model):
         for callback in self.trainer.callbacks:
             callback.log = model.log
             callback.log_dict = model.log_dict
+
+    def attach_model_callbacks(self, model: LightningModule) -> None:
+        """
+        Attaches the callbacks defined in the model.
+        If a callback returned by the model's configure_callback method has the same type as one or several
+        callbacks already present in the trainer callbacks list, it will replace them.
+
+        Args:
+            A model which may or may not define new callbacks in
+            :meth:`~pytorch_lightning.core.lightning.LightningModule.configure_callbacks`.
+        """
+        model_callbacks = model.configure_callbacks()
+        model_callback_types = [type(c) for c in model_callbacks]
+        # remove all callbacks with a type that occurs in model callbacks
```

Comment:
That could be an option. 
However, it would require additional logic when we allow to append without replacement, because we probably want to avoid keep adding callbacks when calling trainer.fit/test multiple times (not idempotent).

Reply:
Good point. Lets hold off on it and see if people request it.